### PR TITLE
Follow up #35217

### PR DESCRIPTION
**Pull request by kaapstorm**
## Technical Summary

Follows up feedback from PR #35217

## Feature Flag

`ARCGIS_INTEGRATION`

### Safety story

Ran tests locally

### Automated test coverage

Test coverage included

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/motech/repeaters/expression/repeaters.py [hunk 1]
```diff
@@ -222,39 +222,37 @@ def send_request(self, repeat_record, payload):
...
         >>> resp.status_code
         503
         >>> resp.reason
         'An error occurred.'
 
         """
-        reason = response_json['error']['message']
-        if 'messageCode' in response_json['error']:
-            reason += f' ({response_json["error"]["messageCode"]})'
+        reason = error_json.get('message', 'Unspecified ArcGIS REST API error')
```
Comment (mkangia) [inline corehq/motech/repeaters/expression/repeaters.py:249]: should we add a message if none if provided?
(I assume RepeaterResponse is what we store and show on HQ as the response received from the external API)
I'd expect HQ to tell what response was received from the API, but if we overwrite that, that could cause confusion.

same for the code below.

Comment (mkangia) [reply corehq/motech/repeaters/expression/repeaters.py:249]: I see your comment here now, https://github.com/dimagi/commcare-hq/pull/35217#discussion_r1800196190
So overriting `code` seems okay. Though if we can add a comment for it in the code, that would be helpful.

File: corehq/motech/repeaters/expression/repeaters.py [hunk 2]
```diff
@@ -246,11 +246,23 @@ def _error_response(error_json):
...
+        # response schema, so we have to guess based on what we've seen.
+
+        # `status_code` is required for us to make decisions about the
+        # repeat record.  If `error_json` does not include "code", then
+        # use 500 so that the repeat record will be sent again later.
+        status_code = error_json.get('code', 500)
+
+        # `reason` is what is shown in the Repeat Records Report under
+        # the "Responses" button. If `error_json` is missing "message",
+        # then set a value that is more useful to users than no message.
```
Comment (mkangia) [inline corehq/motech/repeaters/expression/repeaters.py:259]: Nice comments.

> If `error_json` is missing "message", then set a value that is more useful to users than no message.

How about setting it to something that is true like "Missing error message" or "No error message"?

Comment (mkangia) [reply corehq/motech/repeaters/expression/repeaters.py:259]: haha, may be that just came from personal issues I have reading too much into words. "Unspecified" could mean different for me.
I just thought using a message that users see more often could be useful and hence the suggestion.

File: corehq/motech/repeaters/expression/repeaters.py [hunk 3]
```diff
@@ -222,39 +222,49 @@ def send_request(self, repeat_record, payload):
...
+        reason = error_json.get('message', '[No error message given by ArcGIS]')
```
Comment (mkangia) [inline corehq/motech/repeaters/expression/repeaters.py:260]: nit: You could consider translating it.

**Review by mkangia — APPROVED** [review]

**Comment by mkangia** [comment]
Thanks for the follow up @kaapstorm

**Review by gherceg — APPROVED** [review]

**Pull request merged**